Modules/System/AL Rest Client/src/ALHttpRequestMessage.Codeunit.al

```diff
@@ -0,0 +1,87 @@
+/// <summary>Holder object for the HTTP request data.</summary>
+codeunit 2352 "AL Http Request Message"
+{
+    var
+        HttpRequestMessage: HttpRequestMessage;
+        Initialized: Boolean;
+        NotInitializedErr: Label 'The Http Request Message has been initialized';
+
+    /// <summary>Initializes the HttpRequestMessage object.</summary>
+    procedure Initialize()
+    begin
+        Clear(HttpRequestMessage);
+        Initialized := true;
+    end;
+
+    /// <summary>Sets the HTTP method or the HttpRequestMessage object.</summary>
+    /// <param name="Method">The HTTP method to use. Valid options are GET, POST, PATCH, PUT, DELETE, HEAD, OPTIONS</param>
+    /// <remarks>Default method is GET</remarks>
+    procedure SetHttpMethod(Method: Text)
+    begin
+        AssertInitialized();
+        HttpRequestMessage.Method := Method;
+    end;
+
+    /// <summary>Sets the HTTP method for the HttpRequestMessage object.</summary>
+    /// <param name="Method">The HTTP method to use.</param>
+    /// <remarks>Default method is GET</remarks>
+    procedure SetHttpMethod(Method: Enum "Http Method")
+    begin
+        SetHttpMethod(Method.Names.Get(Method.Ordinals.IndexOf(Method.AsInteger())));
+    end;
+
+    /// <summary>Sets the Uri used for the HttpRequestMessage object.</summary>
+    /// <param name="Uri">The Uri to use for the HTTP request.</param>
+    /// <remarks>The valued must not be a relative URI.</remarks>
+    procedure SetRequestUri(Uri: Text)
+    begin
+        AssertInitialized();
+        HttpRequestMessage.SetRequestUri(Uri);
+    end;
+
+    /// <summary>Adds a request header to the HttpRequestMessage object.</summary>
+    /// <param name="HeaderName">The name of the header to add.</param>
+    /// <param name="HeaderValue">The value of the header to add.</param>
+    /// <remarks>If the header already exists, it will be overwritten.</remarks>
+    [NonDebuggable]
+    procedure AddRequestHeader(HeaderName: Text; HeaderValue: Text)
```

Comment:
`ALHttpContent` has `SetHeader` while `ALHttpRequest` has `AddRequestHeader`, shouldn't they both be `SetHeader`?
I can understand a distinction in `ALHttpClient` like `SetDefaultRequestHeader` but one could argue it should also just be `SetHeader`, since each belongs to their own codeunit.

I could also see some of the helpers setting a specific header like:
`SetUserAgentHeader(HeaderValue)`
But wouldn't this be more scalable?
`SetHeader(HeaderType: Enum "Http Header Type"; HeaderValue: Text)`
Usage:
`SetHeader("Http Header Type"::UserAgent, HeaderValue)`
This way, there's no need for specific helper functions for each known header.

Reply:
The name has now been aligned on Http Content and Http Request Message to be SetHeader. With an explanation that it will set a new value for an existing header or add the header if it does not exists.
The method for the Http Client keep the name SetDefaultRequestHeader to distinguish it from the others. They work different because they are added to every request done with the same Http Client instance.

Not sure about using Enum for known header types. That would involve a huge case statement or an interface and an implementation codeunit for every single enum value to return the correct header name. I think we should keep the current code and see if there is need to come up with an Enum in the future.